src/services/job_service.py

```diff
@@ -30,36 +32,57 @@
 from utils.logging import logger
 from utils.string_helpers import split_address
 
-# TODO: Support cron jobs
-# def is_cron_job(blueprint_ref: str) -> bool:
-#     """Checks if the type SIMOS.CRONJOB is in the list of types the blueprint extends from"""
-#     all_extends = get_extends_from(blueprint_ref)
-#     return SIMOS.CRON_JOB.value in all_extends
 
+class Singleton(type):
+    _instances: Dict[Singleton, Singleton] = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
+
+
+class JobStore(metaclass=Singleton):
+    job_store = redis.Redis(
+        host=config.SCHEDULER_REDIS_HOST,
+        port=config.SCHEDULER_REDIS_PORT,
+        db=0,
+        password=config.SCHEDULER_REDIS_PASSWORD,
+        ssl=config.SCHEDULER_REDIS_SSL,
+        socket_timeout=5,
+        socket_connect_timeout=5,
+    )
 
-def schedule_cron_job(scheduler: BackgroundScheduler, function: Callable, job: Job) -> str:
+
+def get_job_store():
+    return JobStore().job_store
+
+
```

Comment:
Not sure much was gained with this Singleton stuff.

Perhaps just have "get_job_store()" return redis.Redis(...)?